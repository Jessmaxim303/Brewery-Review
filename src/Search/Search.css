.search_box {
	width: 900px;
	height: 400px;
	margin: 12% auto;
	display: flex;
	flex-direction: row;
	align-items: center;
}

.search_logo-box {
	width: 250px;
	height: 250px;
	background: #328cc1;
	margin: 0 5%;
	font-size: 90px;
	color: #f5f5f5;

}

.search_form-box {
	width: 650px;
	height: 250px;
	background: #328cc1;
	display: flex;
	flex-direction: column;
	align-items: center;
	font-size: 20px;
}

.search_button {
	height: 40px;
  width: 355px;
  background: #0b3c5d;
  border: none;
  margin: 15px;
  font-size: 20px;
  color: #f5f5f5;
}

.search_input {
	height: 40px;
  width: 60%;
  margin: 15px;
}

.search_logo-letters {
	margin-top: 38px;
}

.search_input-label {
	margin-top: 20px;
}
