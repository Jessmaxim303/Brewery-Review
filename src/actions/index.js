export const getNews = news => ({
  type: 'GET_NEWS',
  news
});

