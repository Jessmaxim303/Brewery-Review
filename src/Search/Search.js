import React, { Component } from 'react';
import { fetchStateNewsApi } from '../apiCalls/apiCalls.js';
import { connect } from 'react-redux';
import { getNews } from '../actions';
import './Search.css';

import { Link } from "react-router-dom";

export class SearchForm extends Component {
	constructor(props) {
    super(props)
    this.state = {
      area: ''
    }
  }

  componentDidMount() {
    Promise.all([this.loadStateNews()])
  }

  loadStateNews() {
    fetchStateNewsApi('area', this.state.area)
    .then(data => {
      this.props.getNews(data)
    })
    // .then(data => this.setState( data ))
  }

  handleChange = (e) => {
    this.setState({ area: e.target.value });
  }

  loadStateName = () => {
  	console.log(this.state.area)
  	this.props.localNews(this.state.area)
  }

	render() {
		return(
      <form autoComplete="off" className="search_box">
        <section className="search_logo-box">
          <h1 className="search_logo-letters">Lb</h1>
        </section>
        <section className="search_form-box">
          <label for="name" className="search_input-label">Pick a state:</label>
          <input
    	      type="text"
    	      className="search_input"
    	      value={this.state.area}
    	      placeholder="Enter Your State"
    	      name="name"
            onChange={this.handleChange}
    	    />
          <Link to={{ pathname: `/state` }}>
    	        <button className="search_button" onClick={() => this.loadStateNews()}>Find!</button>
          </Link>
        </section>
      </form>
			)
	}

}

// export const mapStateToProps = state => {
//   return (
//   breweries: state
//   )
// }

export const mapDispatchToProps = dispatch => ({
  getNews: (area) => dispatch( getNews(area) )

})

export default connect(null,  mapDispatchToProps)(SearchForm);
