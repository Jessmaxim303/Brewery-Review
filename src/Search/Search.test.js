import React, { Component } from 'react';
import { connect } from 'react-redux';
import { shallow } from 'enzyme';
import { getNews } from '../actions';
import { SearchForm, mapStateToProps, mapDispatchToProps } from './Search.js';
import PropTypes from 'prop-types';


describe('Search', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< SearchForm />)
  })
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an object with the movies and ratings array and a loading status', () => {
      const mockState = {
        movies: [{title: 'Parasite', rating: 9}, {title: '1917', rating: 8}],
        
      }
      const expected = {
        movies: [{title: 'Parasite', rating: 9}, {title: '1917', rating: 8}],
        
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

})
