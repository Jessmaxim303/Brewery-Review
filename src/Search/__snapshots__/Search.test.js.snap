// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `undefined`;

exports[`Search should match a snapshot 1`] = `undefined`;

exports[`SearchForm should match a snapshot 1`] = `undefined`;
